File: library/swift/stats/Gauge.swift
import Foundation

/// A time series gauge.
@objc
public protocol Gauge: AnyObject {
  /// Set the gauge with the given value.
  func set(value: Int)

  /// Add the given amount to the gauge.
  func add(amount: Int)

  /// Subtract the given amount from the gauge.
  func sub(amount: Int)

  /// Gets a gauge with tags attached.
  func attach(tags: [Tag]) -> Gauge
}

File: library/kotlin/io/envoyproxy/envoymobile/stats/Gauge.kt
package io.envoyproxy.envoymobile

/** A time series gauge. */
interface Gauge : Stats {

  /** Sets the gauge by the given value. */
  fun set(value: Int)

  /** Adds the given amount to the gauge. */
  fun add(amount: Int)

  /** Subtracts the given amount from the gauge. */
  fun sub(amount: Int)

  /** Gets a gauge with the list of tags attached. */
  fun attach(tags: List<Tag>): Gauge
}
